azurerm/internal/services/datafactory/data_factory_dataset_binary_resource.go

```diff
@@ -0,0 +1,409 @@
+		}
+	}
+
+	location := expandDataFactoryDatasetLocation(d)
+	if location == nil {
+		return fmt.Errorf("one of `http_server_location`, `azure_blob_storage_location` or `sftp_server_location`, must be specified to create a DataFactory Binary Dataset")
+	}
+
+	binaryDatasetProperties := datafactory.BinaryDatasetTypeProperties{
+		Location: location,
+	}
+
+	if _, ok := d.GetOk("compression"); ok {
+		binaryDatasetProperties.Compression = expandDataFactoryDatasetCompression(d)
+	}
+
+	linkedServiceName := d.Get("linked_service_name").(string)
+	linkedServiceType := "LinkedServiceReference"
+	linkedService := &datafactory.LinkedServiceReference{
+		ReferenceName: &linkedServiceName,
+		Type:          &linkedServiceType,
+	}
+
+	description := d.Get("description").(string)
+	// TODO
+	binaryTableset := datafactory.BinaryDataset{
+		BinaryDatasetTypeProperties: &binaryDatasetProperties,
+		LinkedServiceName:           linkedService,
+		Description:                 &description,
+	}
+
+	if v, ok := d.GetOk("folder"); ok {
+		name := v.(string)
+		binaryTableset.Folder = &datafactory.DatasetFolder{
+			Name: &name,
+		}
+	}
+
+	if v, ok := d.GetOk("parameters"); ok {
+		binaryTableset.Parameters = expandDataFactoryParameters(v.(map[string]interface{}))
+	}
+
+	if v, ok := d.GetOk("annotations"); ok {
+		annotations := v.([]interface{})
+		binaryTableset.Annotations = &annotations
+	}
+
+	if v, ok := d.GetOk("additional_properties"); ok {
+		binaryTableset.AdditionalProperties = v.(map[string]interface{})
+	}
+
+	datasetType := string(datafactory.TypeBasicDatasetTypeBinary)
+	dataset := datafactory.DatasetResource{
+		Properties: &binaryTableset,
+		Type:       &datasetType,
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, dataFactoryName, name, dataset, ""); err != nil {
+		return fmt.Errorf("creating/updating Data Factory Dataset Binary  %q (Data Factory %q / Resource Group %q): %s", name, dataFactoryName, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+	if err != nil {
+		return fmt.Errorf("retrieving Data Factory Dataset Binary %q (Data Factory %q / Resource Group %q): %s", name, dataFactoryName, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("cannot read Data Factory Dataset Binary %q (Data Factory %q / Resource Group %q): %s", name, dataFactoryName, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceDataFactoryDatasetBinaryRead(d, meta)
+}
+
+func resourceDataFactoryDatasetBinaryRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataFactory.DatasetClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	dataFactoryName := id.Path["factories"]
+	name := id.Path["datasets"]
```

Comment:
we could directly use `parse.Dataset(d.Id())`

Reply:
Done